From 9fab8bdaf034c20e985a348613c2111bbeba8b68 Mon Sep 17 00:00:00 2001
From: Lukasz Bartosik <lbartosik@marvell.com>
Date: Fri, 28 Dec 2018 15:22:27 +0300
Subject: [PATCH 0877/1051] crypto: cpt - set PCI bus master flag for CPT

Set PCI bus master flag for both CPT PF and CPT VF devices. This is
to ensure that CPT devices can send in-band interrupts (MSIX).

Change-Id: Ia3cbc0103f41910445732af63daf3e77c31534da
Signed-off-by: Lukasz Bartosik <lbartosik@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/1784
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/crypto/cavium/cpt/9x/cpt9x_pf_main.c | 2 ++
 drivers/crypto/cavium/cpt/9x/cpt9x_vf_main.c | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/drivers/crypto/cavium/cpt/9x/cpt9x_pf_main.c b/drivers/crypto/cavium/cpt/9x/cpt9x_pf_main.c
index c52143a05110..973684dfec2a 100644
--- a/drivers/crypto/cavium/cpt/9x/cpt9x_pf_main.c
+++ b/drivers/crypto/cavium/cpt/9x/cpt9x_pf_main.c
@@ -535,6 +535,8 @@ static int cptpf_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
 		goto cpt_err_set_drvdata;
 	}
 
+	pci_set_master(pdev);
+
 	err = pci_request_regions(pdev, DRV_NAME);
 	if (err) {
 		dev_err(dev, "PCI request regions failed 0x%x\n", err);
diff --git a/drivers/crypto/cavium/cpt/9x/cpt9x_vf_main.c b/drivers/crypto/cavium/cpt/9x/cpt9x_vf_main.c
index 66742ca8cf7b..8e4c55120a58 100644
--- a/drivers/crypto/cavium/cpt/9x/cpt9x_vf_main.c
+++ b/drivers/crypto/cavium/cpt/9x/cpt9x_vf_main.c
@@ -151,6 +151,8 @@ static int cptvf_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
 		goto cpt_err_set_drvdata;
 	}
 
+	pci_set_master(pdev);
+
 	err = pci_request_regions(pdev, DRV_NAME);
 	if (err) {
 		dev_err(dev, "PCI request regions failed 0x%x\n", err);
-- 
2.17.1

